yt/data_objects/data_containers.py

```diff
@@ -257,15 +265,19 @@ def __getitem__(self, key):
         # hanging off the dataset to define this unit object.
         # Note that this is less succinct so that we can account for the case
         # when there are, for example, no elements in the object.
-        rv = self.field_data.get(f, None)
-        if rv is None:
+        try:
```

Comment:
I honestly can't remember why it was this way.  I think this is better, though, and more clear.